src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_pending_resumption_handler.h

```diff
@@ -0,0 +1,173 @@
+    }
+  };
+
+  /**
+   * @brief Handles a successful response from HMI
+   * @param event contains response from HMI
+   * @param module_uid a unique identify a module
+   */
+  void HandleSuccessfulResponse(
+      const application_manager::event_engine::Event& event,
+      const ModuleUid& module_uid);
+
+  /**
+   * @brief Creates next GetInteriorVehicleData
+   * subscription request if a previous resumption of subscriptions is not
+   * successful
+   * @param module a unique identify a module
+   */
+  void ProcessNextFreezedResumption(const ModuleUid& module);
+
+  /**
+   * @brief Notifies subscriber about resumption status
+   * @param subscription_response response from HMI
+   * @param correaltion_id unique ID
+   */
+  void RaiseEventForResponse(
+      const smart_objects::SmartObject& subscription_response,
+      const uint32_t correlation_id) const;
+
+  /**
+   * @brief Checks if sdl is still waiting
+   * responce for specified module
+   * @param module a unique identify a module
+   */
+  bool IsPendingForResponse(const ModuleUid& module) const;
+
+  /**
+   * @brief Checks if any app except passed is subscribed
+   * to a given module
+   * @param module module to check
+   * @param app_id app to ignore subscription
+   * @return true if any app except passed is subscribed to module, otherwise
+   * false
+   */
+  bool IsOtherAppsSubscribed(const uint32_t app_id,
+                             const ModuleUid& module) const;
+
+  /**
+   * @brief Returns pending request, which corresponds to
+   * correlation ID
+   * @param corr_id unique ID
+   * @return optional object, which contains subscription request, or empty
+   * optional, if such request wasn't found
+   */
+  utils::Optional<smart_objects::SmartObject> GetPendingRequest(
+      const uint32_t corr_id);
+
+  /**
+   * @brief Returns ID of application, related to pending request, which
+   * corresponds to correlation ID
+   * @param corr_id unique ID
+   * @return optional object, which contains ID of application, or empty
+   * optional, if such request wasn't found
+   */
+  utils::Optional<uint32_t> GetPendingApp(const uint32_t corr_id);
+
+  /**
+   * @brief Removes pending request, which corresponds to
+   * correlation ID
+   * @param corr_id unique ID
+   */
+  void RemovePendingRequest(const uint32_t corr_id);
+
+  /**
+   * @brief Adds pending request, related to
+   * application ID
+   * @param request_so pending request message
+   */
+  void AddPendingRequest(const uint32_t app_id,
+                         const smart_objects::SmartObject request_so);
+
+  /**
+   * @brief QueueFreezedResumptions contains subscription request,
+   * which will be sent to the HMI, in the case if a previous resumption of
+   * subscriptions for the same module is not successful
+   */
+  using QueueFreezedResumptions = std::queue<PendingRequest>;
+  std::map<ModuleUid, QueueFreezedResumptions> freezed_resumptions_;
```

Comment:
```suggestion
  using PendingRequestQueue = std::queue<PendingRequest>;
  std::map<ModuleUid, PendingRequestQueue> paused_resumptions_;
```
I still think that the typedef is a bit overkill, but at least the name could be made clearer

Reply:
@jacobkeeler Done 07385665dccf8609db73e02f4cacabe73fa699e5